62_CAD/DrawResourcesFiller.cpp

```diff
@@ -218,6 +236,52 @@ void DrawResourcesFiller::drawPolyline(const CPolylineBase& polyline, uint32_t p
 	}
 }
 
+void DrawResourcesFiller::drawTriangleMesh(const CTriangleMesh& mesh, CTriangleMesh::DrawData& drawData, const DTMSettingsInfo& dtmSettingsInfo, SIntendedSubmitInfo& intendedNextSubmit)
+{
+	ICPUBuffer::SCreationParams geometryBuffParams;
+	
+	// concatenate the index and vertex buffer into the geometry buffer
+	const size_t indexBuffByteSize = mesh.getIdxBuffByteSize();
+	const size_t vtxBuffByteSize = mesh.getVtxBuffByteSize();
+	const size_t geometryBufferDataToAddByteSize = indexBuffByteSize + vtxBuffByteSize;
+
+	// copy into gemoetry cpu buffer insteaed
+
+	// TODO: rename, its not just points
+	const uint32_t maxGeometryBufferPoints = static_cast<uint32_t>(maxGeometryBufferSize - currentGeometryBufferSize);
+
+	// TODO: assert of geometry buffer size, do i need to check if size of objects to be added <= maxGeometryBufferPoints?
+	// TODO: auto submit instead of assert
+	assert(geometryBufferDataToAddByteSize <= maxGeometryBufferPoints);
+
+	// TODO: vertices need to be aligned to 8?
```

Comment:
Yes I probably missed this in other places, we need to align to 8 bytes because the vertices have doubles in them. Keep this comment, I'll come back to it